src/test/fuzz/orphanage.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <net_processing.h>
+#include <primitives/transaction.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/script.h>
+#include <test/util/setup_common.h>
+#include <txorphanage.h>
+#include <validation.h>
+
+
+FUZZ_TARGET(orphanage)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    TxOrphanage orphanage;
+    std::set<uint256> orphan_work_set;
+    std::set<COutPoint> outpoints;
+    // initial outpoints used to construct transactions later
+    for (uint8_t i = 0; i < 4; i++) {
+        outpoints.emplace(uint256{i}, 0);
+    }
+
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 2 * DEFAULT_MAX_ORPHAN_TRANSACTIONS)
+    {
+        // construct transaction
+        const CTransactionRef tx = [&] {
+            CMutableTransaction tx_mut;
+            const auto num_in = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, outpoints.size());
+            const auto num_out = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, outpoints.size());
+            // pick unique outpoints from g_outpoints as input
+            for (uint32_t i = 0; i < num_in; i++) {
+                auto pop = outpoints.begin();
+                std::advance(pop, fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(0, outpoints.size() - 1));
+                const auto prevout = *pop;
+                tx_mut.vin.emplace_back(prevout);
+                outpoints.erase(pop);
+            }
+            // output amount will not affect txorphanage
+            for (uint32_t i = 0; i < num_out; i++) {
+                tx_mut.vout.emplace_back(CAmount{0}, P2WSH_OP_TRUE);
+            }
+            // restore previously removed outpoints
```

Comment:
The loop are used to fill the tx's `vout`, you mean it is not necessary since we don't have any code that verifies content of `vout`? Should we replace it with `tx.vout.resize(num_out)` or just keep `vout` empty?

Reply:
The comment actually refers to the loop below. 
```cpp
// restore previously removed outpoints
            for (auto& in : tx_mut.vin) {
                outpoints.insert(in.prevout);
            }
```
Since we want outpoints used by `vin` to be unique, whenever an outpoint is used, it is erased from the set temporarily. After `vin` is constructed, we insert them back to the sets.
The logic here is the same as `tx_pool_standard` target https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/tx_pool.cpp#L205-L208

Reply:
This is because the tx_pool_standard target does the accounting later on, depending on whether the tx was accepted or not https://github.com/bitcoin/bitcoin/blob/cc22bd7f708fc3f5e793bf0138cd340f71c0feb9/src/test/fuzz/tx_pool.cpp#L296

However, your fuzz target does not have a concept of "rejected" transactions right now. Does it?

If you want to "restore" the outpoints, it might be better to not delete them in the first place.